src/main/java/com/bookmanagement/BookManagement/controller/AuthorController.java

```diff
@@ -0,0 +1,50 @@
+package com.bookmanagement.BookManagement.controller;
+
+import com.bookmanagement.BookManagement.dto.AuthorDto;
+import com.bookmanagement.BookManagement.dto.BookDto;
+import com.bookmanagement.BookManagement.mapper.AuthorMapper;
+import com.bookmanagement.BookManagement.repository.AuthorRepository;
+import com.bookmanagement.BookManagement.repository.BookRepository;
+import com.bookmanagement.BookManagement.service.AuthorService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/authors")
+public class AuthorController {
+
+    private AuthorService authorService;
+    private AuthorRepository authorRepository;
+
+    private BookRepository bookRepository;
+    private AuthorMapper authorMapper;
+
+
+    @Autowired
+    public AuthorController(@Qualifier("secondAuthorService") AuthorService authorService) {
+        this.authorService = authorService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<AuthorDto>> getAllAuthors() {
+        authorService.show();
+        return ResponseEntity.ok(authorService.getAllAuthors());
+    }
+
+    // To create Author and Books
+    @PostMapping("/books")
+    public ResponseEntity<AuthorDto> createAuthorBook(@RequestBody AuthorDto authorDto)
```

Comment:
In my opinion, Book and Author are not tightly related. Hence, the book should be created independently of the author
You can add an API over here to map a book to the author.

Reply:
Removed this createAuthorBook and created independently addAuthor, assignAuthorToBook.